### ISSUE TITLE
.jp-mod-running CSS class available for other files types.

### ISSUE THREAD

**Issue by hbcarlos** [jupyterlab/jupyterlab#8649]
### Problem

Hi, I'm developing an extension for jupyterlab that manages a specific filetype from [ROS](http://wiki.ros.org/roslaunch/XML) (we can say like a Dockerfile) and launches these files on the backend using subprocess. I need to track these subprocesses and show his state (is running or not) to the user and the cleanest way to do it is using this green dot that appears before the name of a notebook in the file browser when a notebook has a kernel session running.

The problem is that this CSS class (.jp-mod-running) is removed and added again every time the file browser updates and is only added to these notebooks that have a kernel session running.
The problem comes from [onUpdateRequest()](https://github.com/jupyterlab/jupyterlab/blob/e6b3e4a8987f366a28bfa090e8d2f7d4d1a9ed34/packages/filebrowser/src/listing.ts#L726) method in the DirListing class.

### Proposed Solution

Maybe adding a list of files running in DirListing class, hence, the kernel sessions can keep the same functionality updating this list and other files can be added/removed to this list.

### Additional context

This can be very useful to develop other extensions that need to launch subprocess from a specific filetype or even simple python scripts without opening the file.

**Comment by jasongrout** [comment]
I think you're basically proposing expanding the definition of "running" to more than just the kernels JLab is managing. Another approach would be a general annotation system for annotating files in the file browser with badges, with the "kernel running" badge as just one possibility.

**Comment by hbcarlos** [comment]
Exactly, I think your approach would be better, seams more generic and flexible.

**Comment by SylvainCorlay** [comment]
> expanding the definition of "running" to more than just the kernels JLab is managing

+1 on this.
